services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/Domain.kt

```diff
@@ -0,0 +1,50 @@
+package fi.hel.haitaton.hanke.attachment.common
+
+import fi.hel.haitaton.hanke.allu.Attachment as AlluAttachment
+import fi.hel.haitaton.hanke.allu.AttachmentMetadata
+import java.time.OffsetDateTime
+import java.util.UUID
+
+sealed interface Attachment {
+    val id: UUID
+    val fileName: String
+    val contentType: String
+    val createdByUserId: String
+    val createdAt: OffsetDateTime
+    val blobLocation: String?
+}
+
+data class ApplicationAttachment(
```

Comment:
The stuff in common/AttachmentMetaData should probably be also moved here. Or vice versa. And there may be a need for some renaming.

I like to think that an Attachment should have both metadata and content. So there would be three object types: AttachmentMetadata, AttachmentContent and together they create Attachment. If there is no use cases for having a whole Attachment, we could just have metadata and content separately.

Reply:
The classes in AttachmentMetadata are used as DTOs, where these are purely domain classes. So there's a clear distinction between them. Maybe the difference could be emphasized by renaming AttachmentData.kt to Dto.kt. Or by renaming the AttachmentMetadata classes to AttachmentDto, AttachmentMetadataDto or AttachmentMetadataResponse. Then we could use the AttachmentMetadata name for the domain class.

Reply:
The `AttachmentContent` and `HankeAttachmentWithContent` are used as domain classes. I'll move them to Domain.kt.

Reply:
Should we use separate classes instead of Domain.kt? Maybe `domain` package?

Reply:
In https://github.com/City-of-Helsinki/haitaton-backend/pull/545/commits/da7c583de17ce747acf6cd7031b70bde8ca02872, I renamed:
- HankeAttachment to HankeAttachmentMetadataDto
- ApplicationAttachmentMetadata to ApplicationAttachmentMetadataDto
- ApplicationAttachment to ApplicationAttachmentMetadata
- AttachmentMetadata.kt-file to `Dto.kt`

I also moved AttachmentContent and HankeAttachmentWithContent to `Domain.kt`.

Is this ok?